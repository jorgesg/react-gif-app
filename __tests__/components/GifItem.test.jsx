import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Testing <GifItem />', () => {

    const title = 'Gif title test';
    const url = 'https://giftest.com/url-test-1234'

    test('Match snapshot', () => {
        const { container } = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('Should display image with correct url and alt', () => {
        render(<GifItem title={title} url={url}/>);
        
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Should display title', () => {
        render(<GifItem title={title} url={url}/>);

        expect(screen.getByText(title)).toBeTruthy();
    })

})