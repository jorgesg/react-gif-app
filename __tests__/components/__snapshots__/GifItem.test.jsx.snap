// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> Match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif title test"
      src="https://giftest.com/url-test-1234"
    />
    <p>
      Gif title test
    </p>
  </div>
</div>
`;
